src/ja/raw18/src/eu/kanade/tachiyomi/extension/ja/raw18/Raw18.kt

```diff
@@ -0,0 +1,127 @@
+package eu.kanade.tachiyomi.extension.ja.raw18
+
+import eu.kanade.tachiyomi.multisrc.wpcomics.WPComics
+import eu.kanade.tachiyomi.network.GET
+import eu.kanade.tachiyomi.source.model.FilterList
+import eu.kanade.tachiyomi.source.model.SChapter
+import eu.kanade.tachiyomi.source.model.SManga
+import okhttp3.HttpUrl.Companion.toHttpUrl
+import okhttp3.Request
+import org.jsoup.nodes.Document
+import org.jsoup.nodes.Element
+import java.text.SimpleDateFormat
+import java.util.Calendar
+import java.util.Locale
+
+class Raw18 : WPComics("Raw18", "https://raw18.net", "ja", SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSSSSS'Z'", Locale.JAPANESE), null) {
+    override val searchPath = "search/manga"
+    override val genresUrlDelimiter = "="
+    override fun popularMangaSelector() = "div.items article.item"
+    override fun popularMangaNextPageSelector() = "li:nth-last-child(2) a.page-link"
+    override fun searchMangaSelector() = popularMangaSelector()
+    override fun mangaDetailsParse(document: Document): SManga {
+        return SManga.create().apply {
+            document.select("article#item-detail").let { info ->
+                author = info.select("li.author p.col-xs-8").text()
+                status = when {
+                    info.select("li.status p.col-xs-8").text().contains("Ongoing", true) -> SManga.ONGOING
```

Comment:
Assuming you expect more status values, I would move parts of it into a variable.

```kt
val x = info.select("li.status p.col-xs-8").text()
```

Reply:
opted for toStatus() from wpcomics